flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTerminationTest.java

```diff
@@ -227,21 +229,13 @@ protected void cleanup() throws Exception {
 			// has been stopped
 			CLEANUP_LATCH.trigger();
 
-			// wait until handle async exception has been called to proceed with the termination of the
-			// StreamTask
-			HANDLE_ASYNC_EXCEPTION_LATCH.await();
+			// wait until all async checkpoint threads are terminated, so that no more exceptions can be reported
+			Assert.assertTrue(getAsyncOperationsThreadPool().awaitTermination(30L, TimeUnit.SECONDS));
```

Comment:
Where do we trigger the shut down of the `getAsyncOperationsThreadpool`?

Reply:
I happens through the StreamTask, before cleanups.